### ISSUE TITLE
Methodology?

### ISSUE THREAD

**Issue by boscacci** [learn-co-students/dsc-3-32-11-xgboost-lab-nyc-career-ds-102218#1]
`grid_clf = GridSearchCV(clf, param_grid, scoring='accuracy', cv=None, n_jobs=1)
grid_clf.fit(scaled_df, labels)

best_parameters = grid_clf.best_params_

print("Grid Search found the following optimal parameters: ")
for param_name in sorted(best_parameters.keys()):
    print("%s: %r" % (param_name, best_parameters[param_name]))

training_preds = grid_clf.predict(X_train)
val_preds = grid_clf.predict(X_test)
training_accuracy = accuracy_score(y_train, training_preds)
val_accuracy = accuracy_score(y_test, val_preds)`

Why are we training the grid search on the whole dataset, then trying to validate against two subsets of that same data-set? Isn't this sort of uninformative? Shouldn't we be training the grid search on training/validation data, then testing the predictive power on the test set?
